--- toolchain/cc_toolchain_config.bzl.tpl	2019-05-16 03:44:32.000000000 +0000
+++ toolchain/cc_toolchain_config.bzl.tpl	2019-07-27 05:32:42.953936974 +0000
@@ -132,13 +132,13 @@

     if ctx.attr.cpu == "k8":
         linker_flags = [
-            # Use the lld linker.
-            "-fuse-ld=lld",
+            # Use the gold linker.
+            "-fuse-ld=gold",
             # The linker has no way of knowing if there are C++ objects; so we always link C++ libraries.
             "-L%{toolchain_path_prefix}/lib",
-            "-l:libc++.a",
-            "-l:libc++abi.a",
-            "-l:libunwind.a",
+            # Build with libstdc++ on linux
+            "-lstdc++",
+            "-lgcc_s",
             # Compiler runtime features.
             "-rtlib=compiler-rt",
             # To support libunwind.
@@ -262,7 +262,7 @@
             ),
             flag_set(
                 actions = all_cpp_compile_actions,
-                flag_groups = [flag_group(flags = ["-std=c++17", "-stdlib=libc++"])],
+                flag_groups = [flag_group(flags = ["-std=c++11", "-stdlib=libstdc++" if ctx.attr.cpu == "k8" else "-stdlib=libc++"])],
             ),
         ],
     )
